### PR TITLE
Ps 86

### PR DESCRIPTION
**Pull request by Phmank**
- Added a function to define a start- and enddate
- Added a function to define a pages whitelist that enables the maintenance mode only for these pages

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,6 +24,10 @@ Tab and hold the finger down on the maintenance page for 5s to hide it.
 - `androidLink`: Define an external url for Android devices. A button will be displayed that is linked to the external url.
 - `iosButtonText`: Define a text for the link on iOS devices.
 - `androidButtonText`: Define a text for the link button on Android devices.
+- `startDate`: Define a start date for the maintenance mode. You can also only define a start or an end date. Format: YYYY/MM/DD - HH:mm (e.g. `"2023/04/20 - 00:00"`).
+- `endDate`: Define a end date for the maintenance mode. You can also only define a start or an end date. Format: YYYY/MM/DD - HH:mm (e.g. `"2023/05/17 - 10:00"`).
+- `maintenancePagesWhitelist`: Whitelist that enables the maintenance mode only for configured pages (e.g. `["/cart"]`).
```
Comment (patrick-scherer) [inline README.md:29]: `/item`, `/page` or `/category/373036` does not work - which pages can be added?

File: frontend/portals/MaintenanceMode/index.jsx [hunk 2]
```diff
@@ -85,63 +91,103 @@ class MaintenanceMode extends Component {
...
+
+    return new Date(parseStartDate) < now && new Date(parseEndDate) > now;
+  };
+
+   /**
+   * Checks if there is a page whitelist and only enables maintenance for these pages.
+   * @param {Object} currentRoute App version
+   * @returns {boolean}
+   */
+   hasPageWhiteList = currentRoute => maintenancePagesWhitelist.includes(currentRoute.pattern) ||
```
Comment (patrick-scherer) [inline frontend/portals/MaintenanceMode/index.jsx]: `hasPageWhiteList` = true, when the whitelist is empty

I think the variable should have an other name?

File: extension-config.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "version": "2.1.0-rc.1",
+  "version": "1.4.0",
```
Comment (patrick-scherer) [inline extension-config.json:2]: last version in dev center is 2.2.0-rc.2

File: README.md (hunk 1 above)
Comment (Phmank) [reply README.md:29]: reworked logic

File: frontend/portals/MaintenanceMode/index.jsx (hunk 2 above)
Comment (Phmank) [reply frontend/portals/MaintenanceMode/index.jsx]: renamed

File: extension-config.json (hunk 3 above)
Comment (Phmank) [reply extension-config.json:2]: continued github version

**Pull request merged**